SwiftCompilerSources/Sources/Optimizer/ModulePasses/MandatoryPerformanceOptimizations.swift

```diff
@@ -32,6 +32,11 @@ let mandatoryPerformanceOptimizations = ModulePass(name: "mandatory-performance-
   var worklist = FunctionWorklist()
   worklist.addAllPerformanceAnnotatedFunctions(of: moduleContext)
   worklist.addAllAnnotatedGlobalInitOnceFunctions(of: moduleContext)
+  // For embedded Swift, optimize all the functions (there cannot be any
+  // generics, type metadata, etc.)
+  if moduleContext.options.enableEmbeddedSwift {
+    worklist.addAllNonGenericFunctions(of: moduleContext)
+  }
```

Comment:
`addAllPerformanceAnnotatedFunctions` and `addAllAnnotatedGlobalInitOnceFunctions` should be called in the else-branch of this if-statement